### DISCUSSION TITLE
TypeMismatch when trying to update an Set String

### DISCUSSION THREAD

**Discussion by nimit2801** [dynamoose/dynamoose#1512] [Q&A]
### I'm trying to update set string on DynamoDB with Dynamoose, both of these methods are documented in Dynamoose documentation.
Schema.ts
```ts
const UserSchema = new Dynamoose.Schema(
    {
        uuid: { type: String, hashKey: true, required: true },
        sortkey: { type: String, rangeKey: true, required: true },
 
        petPhotosURI: {
            type: Array,
            schema: [String],
        },
        likedMe: {
            type: Set,
            schema: [String],
        },
    },
    {
        timestamps: true,
    },
);
```
Code where the error is throw
### First Case
```ts
const myId: string = body.myId;

const likedUser = await User.update({ uuid: likedId, sortkey: 'pet/Bruno' }, { $ADD: { likedMe: [myId] } });
```
catches this error
```bash
TypeMismatch: Expected likedMe to be of type string set, instead found type object.
```

### Second Case
```ts
const myId: string = body.myId;

const likedUser = await User.update({ uuid: likedId, sortkey: 'pet/Bruno' }, { $ADD: { likedMe: `${myId}` } });
```

catches the same error
```bash
TypeMismatch: Expected likedMe to be of type string set, instead found type object.
```
### How can I either bypass this error or use the Dynamoose Set String type to eliminate this one

**Comment by fishcharlie** [comment]
@nimit2801 What happens if you try to pass in a set?

**Reply by nimit2801** [reply]
Hey there @fishcharlie same thing happens it gives Object was passed. Apparently `Set<string>` also has `Object` type

**Reply by fishcharlie** [reply]
@nimit2801 And does this work outside of `update`? As in, if you do `create` does it work?

**Reply by nimit2801** [reply]
Let me check, because I'm using the directly this attribute `likedMe` in `model.update`. Will get back to you.

**Reply by nimit2801** [reply]
Also I'm only getting this error when I'm trying to test with Jest! normally if I'm trying to access the Dynamoose API. Not getting this error.

**Reply by nimit2801** [reply]
@fishcharlie getting the same error with `create`

**Reply by nimit2801** [reply]
@fishcharlie any updates on this?

**Reply by fishcharlie** [reply]
@nimit2801 

> Also I'm only getting this error when I'm trying to test with Jest! normally if I'm trying to access the Dynamoose API. Not getting this error.

That normally indicates something outside of Dynamoose and your testing setup. I can't really help with regards to that.

I mean just logically speaking, if it works in your application itself, but doesn't work in Jest, that doesn't really indicate a Dynamoose problem at all.

That being said, the error is for sure coming from Dynamoose.

I'd work on debugging this by isolating further what is going on. Why is the test doing something different than your application itself?

Maybe start by creating a new project that is super simple to reproduce the issue.

If you get to the point where you have a consistent MCVE, feel free to create an issue on this repo.

**Reply by nimit2801** [reply]
@fishcharlie 

https://github.com/nimit2801/Dynamoose-MCVE
This is not even using JEST but facing the same error
#### Error

![image](https://user-images.githubusercontent.com/53462249/190871216-6a82c1f3-fb8e-4a6a-a5a5-f6786fcf766d.png)

**Reply by nimit2801** [reply]
I don't think so this a JEST error. While any request is going through `APIGATEWAY` and then intercepting the `lamda` this error is coming. I don't know is this normal behavior or I'm doing something dumb.

**Reply by fishcharlie** [reply]
@nimit2801 Is API Gateway necessary to reproduce the error? This doesn't seem like an MCVE...

**Reply by nimit2801** [reply]
Yes, without introducing `APIGATEWAY` you can't replicate the error. Normally it's working ig. Maybe my understanding of `MCVE` is wrong then.

**Reply by nimit2801** [reply]
@fishcharlie should I update something here?

**Reply by fishcharlie** [reply]
@nimit2801 You can read more about MCVE at:

- https://github.com/dynamoose/dynamoose/blob/main/CONTRIBUTING.md#minimal-complete-and-verifiable-example-mcve
- https://stackoverflow.com/help/minimal-reproducible-example

The example you provided is not an MCVE. An MCVE is supposed to be a **small** code example that reproduces the issue. It should only include the MINIMAL amount of code necessary to reproduce the issue.

---

> Yes, without introducing APIGATEWAY you can't replicate the error

The fact that you said this, means it's not a Dynamoose error.
